### PR TITLE
Improved better tooltips

### PR DESCRIPTION
**Pull request by C10udburst**
image
image
image
image

### CODE REVIEW

**Comment by seasnail8169** [comment]
can you make the book render with the background like it used to and like shulkers do

**Comment by seasnail8169** [comment]
and same with map

File: src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @Override
+    public int getHeight() {
+        double scale = Modules.get().get(BetterTooltips.class).mapsScale.get();
+        return (int)((128+16)*scale) + 2;
+    }
+
+    @Override
+    public int getWidth(TextRenderer textRenderer) {
+        double scale = Modules.get().get(BetterTooltips.class).mapsScale.get();
+        return (int)((128+16)*scale);
```
Comment (seasnail8169) [inline src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java:40]: a

File: src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private final int mapId;
+
+    public MapTooltipComponent(int mapId) {
+        this.mapId = mapId;
+    }
+
+    @Override
+    public int getHeight() {
+        double scale = Modules.get().get(BetterTooltips.class).mapsScale.get();
+        return (int)((128+16)*scale) + 2;
```
Comment (seasnail8169) [inline src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java:34]: a

File: src/main/java/meteordevelopment/meteorclient/utils/tooltip/BookTooltipComponent.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Override
+    public void drawItems(TextRenderer textRenderer, int x, int y, MatrixStack matrices, ItemRenderer itemRenderer, int z, TextureManager textureManager) {
+        // Background
+        RenderSystem.setShader(GameRenderer::getPositionTexShader);
+        RenderSystem.setShaderTexture(0, TEXTURE_BOOK_BACKGROUND);
+        DrawableHelper.drawTexture(matrices, x, y, z, 12, 0, 112, 134, 179, 179);
+
+        // Content
+        matrices.push();
+        matrices.translate(x+16, y+12, z+1);
```
Comment (seasnail8169) [inline src/main/java/meteordevelopment/meteorclient/utils/tooltip/BookTooltipComponent.java:53]: a

File: src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public TooltipComponent getComponent() { return this; }
+
+    @Override
+    public void drawItems(TextRenderer textRenderer, int x, int y, MatrixStack matrices, ItemRenderer itemRenderer, int z, TextureManager textureManager) {
+        double scale = Modules.get().get(BetterTooltips.class).mapsScale.get();
+
+        // Background
+        matrices.push();
+        matrices.translate(x, y, z);
+        matrices.scale((float)(scale)*2, (float)(scale)*2, 0);
```
Comment (seasnail8169) [inline src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java:53]: a

File: src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        matrices.scale((64+8)/64f,(64+8)/64f,0);
```
Comment (seasnail8169) [inline src/main/java/meteordevelopment/meteorclient/utils/tooltip/MapTooltipComponent.java:54]: a

**Review by seasnail8169 — CHANGES_REQUESTED** [review]
please use le space bar

**Comment by blendifyClient** [comment]
i love this

**Comment by C10udburst** [comment]
> i love this

thank you RectumClient, very insightful

**Pull request merged**